creators/Fenix/StopsCreatorFenix.py

```diff
@@ -1,19 +1,142 @@
 # coding=utf-8
 
+import overpy
+import sys
+
+from transitfeed import util
 from creators.StopsCreator import StopsCreator
+from osmhelper.osm_routes import Route, RouteMaster
+from osmhelper.osm_stops import Stop
 
 
 class StopsCreatorFenix(StopsCreator):
 
     def add_stops_to_schedule(self, schedule, data):
 
-        stops = data.stops
+        # Get stops information
+        stops = data.get_stops()
 
         # add all stops to GTFS
         for stop in stops.values():
+
+            # If there is no name, query one intelligently from OSM
+            if stop.name == Stop.NO_NAME:
+                self._get_interim_stop_name(stop)
+                print stop
```

Comment:
Yes, actually this is the reason why I moved (recently) the cache into it's own class, so we can easily cache it here. I add a line to the `StopCreatorFenix`

```
if stops_changed:
     # Cache stops with newly created stop names
     Cache.write_data('stops', self.stops)
```